src/explore-education-statistics-common/src/modules/charts/components/utils/getReferenceLineLabelPosition.ts

```diff
@@ -9,29 +9,37 @@ interface LabelPosition {
 export default function getReferenceLineLabelPosition({
   axis,
   axisType,
+  lineIsBetweenTwoPoints = false,
```

Comment:
Flags toggling behaviour like this is a bit of a code smell as it adds additional complexity.

I'm not sure we'd need most of the changes in this file if we implement the bounded references lines on the correct axis.